src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs

```diff
@@ -359,6 +382,27 @@ public static IServiceCollection AddMvcLocalization([NotNull] this IServiceColle
             return services;
         }
 
+        // Adds a service if the service type and implementation type hasn't been added yet. This is needed for
+        // services like IConfigureOptions<MvcOptions> or IApplicationModelProvider where you need the ability
+        // to register multiple implementation types for the same service type.
+        private static bool TryAddMultiRegistrationService(IServiceCollection services, ServiceDescriptor descriptor)
```

Comment:
@IDisposable I agree the method name should change if @rynowak follows my suggestion.